server/graphql/asset/src/types.rs

```diff
@@ -0,0 +1,175 @@
+use async_graphql::*;
+use graphql_core::generic_filters::{DateFilterInput, EqualFilterStringInput, StringFilterInput};
+use graphql_core::simple_generic_errors::NodeError;
+use repository::assets::asset::AssetSortField;
+use repository::{
+    assets::asset::{Asset, AssetFilter, AssetSort},
+    EqualFilter,
+};
+use repository::{DateFilter, StringFilter};
+use service::{usize_to_u32, ListResult};
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(rename_items = "camelCase")]
+pub enum AssetSortFieldInput {
+    Name,
+    SerialNumber,
+    InstallationDate,
+    ReplacementDate,
+    ModifiedDatetime,
+}
+#[derive(InputObject)]
+pub struct AssetSortInput {
+    /// Sort query result by `key`
+    key: AssetSortFieldInput,
+    /// Sort query result is sorted descending or ascending (if not provided the default is
+    /// ascending)
+    desc: Option<bool>,
+}
+
+#[derive(InputObject, Clone)]
+pub struct AssetFilterInput {
+    pub name: Option<StringFilterInput>,
+    pub code: Option<StringFilterInput>,
+    pub id: Option<EqualFilterStringInput>,
+    pub serial_number: Option<StringFilterInput>,
+    pub class_id: Option<EqualFilterStringInput>,
+    pub category_id: Option<EqualFilterStringInput>,
+    pub type_id: Option<EqualFilterStringInput>,
+    pub catalogue_item_id: Option<EqualFilterStringInput>,
+    pub installation_date: Option<DateFilterInput>,
+    pub replacement_date: Option<DateFilterInput>,
+}
+
+impl From<AssetFilterInput> for AssetFilter {
+    fn from(f: AssetFilterInput) -> Self {
+        AssetFilter {
+            name: f.name.map(StringFilter::from),
+            code: f.code.map(StringFilter::from),
+            id: f.id.map(EqualFilter::from),
+            serial_number: f.serial_number.map(StringFilter::from),
+            class_id: f.class_id.map(EqualFilter::from),
+            category_id: f.category_id.map(EqualFilter::from),
+            type_id: f.type_id.map(EqualFilter::from),
+            catalogue_item_id: f.catalogue_item_id.map(EqualFilter::from),
+            installation_date: f.installation_date.map(DateFilter::from),
+            replacement_date: f.replacement_date.map(DateFilter::from),
+        }
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub struct AssetNode {
+    pub asset: Asset,
+}
+
+#[derive(SimpleObject)]
+pub struct AssetConnector {
+    total_count: u32,
+    nodes: Vec<AssetNode>,
+}
+
+#[Object]
+impl AssetNode {
+    pub async fn id(&self) -> &str {
+        &self.row().id
+    }
+
+    pub async fn store_id(&self) -> &Option<String> {
+        &self.row().store_id
+    }
+
+    pub async fn name(&self) -> &str {
+        &self.row().name
+    }
+
+    pub async fn code(&self) -> &str {
+        &self.row().code
+    }
+
+    pub async fn serial_number(&self) -> &Option<String> {
+        &self.row().serial_number
+    }
+
+    // TODO: Loaders for store, class, category, type, catalogue_item
```

Comment:
@mark-prins do you need this for frontend? Should we create a new issue for it in github?

Reply:
Yes please!
I've just added an issue (#3103) for class, category and type. There is already a store loader - can we add `catalogue_item` to the issue?